modules/store/src/selector.ts

```diff
@@ -20,10 +20,15 @@ export type DefaultProjectorFn<T> = (...args: any[]) => T;
 export interface MemoizedSelector<
   State,
   Result,
-  ProjectorFn = DefaultProjectorFn<Result>
+  ProjectorFn = DefaultProjectorFn<Result>,
+  ProjectorArgs extends unknown[] = unknown[]
 > extends Selector<State, Result> {
   release(): void;
-  projector: ProjectorFn;
+  projector(...args: ProjectorArgs): Result;
+  /** @deprecated Only use valid arguments for the selector's projector */
+  projector(
+    ...args: ProjectorFn extends (...args: infer Args) => unknown ? Args : any[]
+  ): ProjectorFn extends (...args: unknown[]) => infer R ? R : Result;
```

Comment:
When invalid arguments are used, then the deprecated signature is used. The deprecated signature is backwards compatible with existing usages of the `ProjectorFn` generic parameter by using its parameters and return value in the type.